core/field_angle.js

```diff
@@ -347,10 +423,104 @@ Blockly.FieldAngle.prototype.doClassValidation_ = function(newValue) {
   if (n < 0) {
     n += 360;
   }
-  if (n > Blockly.FieldAngle.WRAP) {
+  if (n > this.getWrap()) {
     n -= 360;
   }
   return n;
 };
 
+/**
+ * Set which direction should make the graphical angle editor increase.
+ * @param {?boolean} clockwise Whether the graphical angle editor should
+ *    increase as it is moved clockwise (true) or counterclockwise (false)
+ *    or the global setting (null).
+ * @return {!Blockly.FieldAngle} Returns itself (for method chaining).
+ */
+Blockly.FieldAngle.prototype.setClockwise = function(clockwise) {
```

Comment:
@RoboErikG thoughts on returning itself for method chaining?  It's a reasonable pattern, but 1) we don't use it anywhere else and 2) we probably want to be consistent.

Reply:
It's nice for convenience, but it's pretty non-standard for instances. I think I'd rather start using an opt_config parameter with a map of the configuration options. The keys should be the same as the JSON config which will also let you use the same code for both.

Reply:
Do you want me to go ahead and update it? or are you waiting for feedback from Rachel?

Reply:
+1 to Erik.  Go ahead and update it.

Reply:
I went to go update this, and then I realized I had more questions and wasn't sure how far reaching the change should be. I [created an issue](https://github.com/google/blockly/issues/2722) to make discussion easier.

Reply:
If you're happy with the clarification on #2722, then just remove the method chaining and this will be good to go.

Reply:
I was planning on getting the abstract field changes in first (just to make sure how I'm thinking about it in my head is actually going to work). But don't worry I haven't forgotten about this little guy!

Also atm #2722 has this guy down as no hooks. This is b/c, for example,  changing the WRAP value at runtime means the field would have to do some extra work recalculating it's value. I think I could get it to work though.

Did you want hooks for the angle field or no?